matchtree.go

```diff
@@ -213,29 +210,29 @@ func (t *symbolSubstrMatchTree) prepare(doc uint32) {
 		fileStart = t.fileEndRunes[doc-1]
 	}
 
-	for len(t.sections) > 0 && t.sections[0].Start < fileStart {
-		t.sections = t.sections[1:]
-		t.secID++
+	var sections []DocumentSection
+	if len(t.sections) > 0 {
+		sections = t.sections[t.fileEndSymbol[doc]:t.fileEndSymbol[doc+1]]
```

Comment:
does doc+1 always exist?

Reply:
Yes, `len(t.fileEndSymbol)` will always be one greater than the number of documents.